gateway-sp-comms/src/single_sp.rs

```diff
@@ -793,6 +796,22 @@ impl SingleSp {
 
         result.result.and_then(expect_caboose_value)
     }
+
+    pub async fn read_sensor_value(&self, id: u32) -> Result<SensorReading> {
+        let v = self
+            .rpc(MgsRequest::ReadSensor(SensorRequest::LastReading { id }))
+            .await
+            .and_then(expect_read_sensor)?;
+        match v {
+            SensorResponse::LastReading(r) => Ok(r),
+            // TODO: Is this a bad overloading of
+            // CommunicationError::BadResponseType?
```

Comment:
I don't think so. `BadResponseType` should never happen, assuming a well-behaved / correct SP, which also seems true here, right? A `LastReading` request should always get a `LastReading` response.

Should we have other methods for sending the other kinds of `SensorRequest`s?

Reply:
> I don't think so. BadResponseType should never happen, assuming a well-behaved / correct SP, which also seems true here, right? A LastReading request should always get a LastReading response.

Yeah, I was wondering whether we wanted a specialized `SpError::Sensor(SensorError::BadResponseType { ... })`, since the reply / response is nested.  I'm fine keeping it as-is, though, since you don't object!

> Should we have other methods for sending the other kinds of SensorRequests?

Eh, I was planning to add them as needed 🤷🏻‍♂️ 

(Writing the extra request kinds was mostly future-proofing, since now we could hypothetically extend `MgsRequest::ReadSensor` without creating more top-level `MgsRequest` variants)